ui/src/join_game/api.ts

```diff
@@ -56,6 +56,49 @@ export function startGame(game_handle: string) {
   };
 }
 
+export function useEscalator(
+  game_state: GameState,
+  connection_status: ConnectionStatus,
+  heister_selected_keyboard: number // HeisterColor
+) {
+  return async (dispatch) => {
+    if (
+      game_state === null ||
+      connection_status !== ConnectionStatus.Connected
+    ) {
+      console.error("Tried to move heister with no game state / connection");
+      return;
+    }
+    var heisters = game_state.getHeistersList();
+    var heister = heisters.find(
+      (h) => h.getHeisterColor() == heister_selected_keyboard
+    );
+    if (heister === undefined) {
+      console.error("Could not find information for heister");
+      return;
+    }
+    var current_position = heister.getMapPosition();
+    if (current_position === undefined) {
+      console.error("Tried to move heister with no position");
+      return;
+    }
+
+    // prep checks done - now let's actually get the dest position, then do the move!
+    let map = game_state.getPossibleEscalatorsMap();
+    var esc_dest = map.get(heister_selected_keyboard);
+    if (esc_dest === undefined) {
+      console.error(
```

Comment:
Maybe just log, since it isn't really an error, just a player trying to do something invalid. If we add client side move validation, this will become an error here.

Reply:
Oh yeah. Ok. (y)